src/base/bittorrent/session.cpp

```diff
@@ -724,6 +770,45 @@ void Session::setSubcategoriesEnabled(bool value)
     emit subcategoriesSupportChanged();
 }
 
+QSet<QString> Session::tags() const
+{
+    return m_tags;
+}
+
+bool Session::isValidTag(const QString &tag)
+{
+    return !tag.trimmed().isEmpty() && !tag.contains(",");
```

Comment:
Shouldn't we have a more strict rule for tags? I would allow only alphanumeric characters and perhaps spaces. Also I would limit the maximum length of the tag.